src/java/org/apache/cassandra/tools/nodetool/Snapshot.java

```diff
@@ -63,6 +67,13 @@ public void execute(NodeProbe probe)
 
             Map<String, String> options = new HashMap<String,String>();
             options.put("skipFlush", Boolean.toString(skipFlush));
+            if (null != ttl) {
+                Duration d = Duration.from(ttl);
+                // ttl for snapshot must be at least 1 minute
+                if (d.getMonths() == 0 && d.getDays() == 0 && d.getNanoseconds() < d.NANOS_PER_MINUTE)
```

Comment:
I think this is just enough:

        if (ttl.toMinutes() < 1)
            throw  ...

Reply:
@pauloricardomg told me to use this class:
https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/Duration.java

It does not supply such method.